### DISCUSSION TITLE
Why the errors of useFormState didn't update?

### DISCUSSION THREAD

**Discussion by leslie555** [react-hook-form/react-hook-form#9680] [Q&A]
```
    const { control, handleSubmit, setValue, watch } = useForm<AddUserFormInputs>({
      mode: 'onBlur',
      defaultValues: defaultValue,
    });
    const { isValid, errors } = useFormState({
      control,
    });
   console.log(isValid, errors);  // false {}
```

This is multi-selector code.
```
<FormSelect
            control={control}
            label="Buildings"
            name="buildings"
            options={accountBuildings}
            mode="multi"
            rules={{
              // required: true,
              validate: {
               // type of v: string[]
                value: (v) => v.length > 0 || 'should select at least one option',
              },
            }}
          />
```
In my case, when I clear the content of my multi-selector, isValid will be false but errors still keep an empty object.

This is codesandbox link:  https://codesandbox.io/s/twilight-field-p3z7x7.
I'm not sure why the `formState.defaultValues` in formSelect.jsx is different from `defaultValue` in form.jsx.

**Comment by leapful** [comment]
@leslie555 

Please provide a codesandbox url to re-produce the issue.

**Reply by leslie555** [reply]
Hi, I added a link, please take a look.

**Reply by leapful** [reply]
@leslie555 :

You are using mode `onBlur` on `useForm` hook therefore the validation only occurs on blur but then when clicking on remove icon of the multiple select component, there is no blur because there is no default focus on your field.

Just set focus on your field first so in case the selected value exists, click on remove icon will trigger validation. Another solution is to change mode `onChange`, like this example: https://codesandbox.io/s/react-hook-form-1-forked-ccjdw3 .
